### PR TITLE
Add tests for chatbot scroll controls & duplicate file upload validation for attachments

### PR DESCRIPTION
**Pull request by HusneShabbir**
**Description:**

This pull request enhances the test coverage of the application with two main additions:

1. **Chatbot Conversation Scroll Controls Test:**

   * Verifies the presence and functionality of "Jump top" and "Jump bottom" buttons in the chatbot conversation.
   * Sends a sample message about OpenShift deployment and checks loading indicators, menu items, and correct scroll behavior.
   * Confirms that the chatbot response contains expected OpenShift deployment details.

2. **File Attachment Validation Tests:**

   * Validates file uploads for various test files (e.g., `package.json` ).
   * Checks successful uploads for supported file types.
   * Implements a `reUploadfile` helper function to verify that uploading a duplicate file triggers a visible alert with the message "File already exists."
   * Confirms that unsupported file types show upload failure messages accordingly.

These additions improve reliability for both chatbot UI interactions and file attachment handling, ensuring a better user experience and robust error handling.

**Testing:**
All tests were executed successfully, ensuring the functionality works as intended.
Screenshot 2025-06-03 at 6 00 05 PM

**Trace:**
[reUploadfile_Trace.zip](https://github.com/user-attachments/files/20574579/reUploadfile_Trace.zip)

[verifyScrollContols_Trace.zip](https://github.com/user-attachments/files/20574581/verifyScrollContols_Trace.zip)

### CODE REVIEW

File: workspaces/lightspeed/packages/app/e2e-tests/utils/fileUpload.ts [hunk 1]
```diff
@@ -36,6 +36,14 @@ export async function uploadFile(page: Page, filePath: string) {
   await fileChooser.setFiles(filePath);
 }
 
+export async function reUploadfile(page: Page, filePath: string) {
+  await uploadFile(page, filePath);
+  await expect(
+    page.getByRole('heading', { name: 'File upload failed' }),
+  ).toBeVisible();
+  await expect(page.getByText('File already exists.')).toBeVisible();
+}
```
Comment (karthikjeeyar) [inline workspaces/lightspeed/packages/app/e2e-tests/utils/fileUpload.ts:45]: The name of this function doesn't describe it's behaviour. It sounds like a re-upload, but the function's logic is actually about verifying that uploading an existing file results in an error. So the pre-requisite for this function to work correctly is to upload the same file before calling this function, I would suggest choose a better name or even better you can warp them into meaningful helper function.

```ts
export async function uploadAndAssertDuplicate(page: Page, filePath: string, fileName: string) {
  await validateSuccessfulUpload(page, fileName);
  await uploadFile(page, filePath);
  await expect(
    page.getByRole('heading', { name: 'File upload failed' }),
  ).toBeVisible();
  await expect(page.getByText('File already exists.')).toBeVisible()
}
```

**Comment by HusneShabbir** [comment]
@karthikjeeyar, @jrichter1
Could you please take a moment to review the changes and let me know if anything needs to be revised or updated further?

File: workspaces/lightspeed/packages/app/e2e-tests/utils/sidebar.ts [hunk 2]
```diff
@@ -49,3 +49,11 @@ export async function assertDrawerState(page: Page, state: 'open' | 'closed') {
     await expectations[state](locator);
   }
 }
+
+export async function sidePanelAssertions(page: Page) {
```
Comment (jrichter1) [inline workspaces/lightspeed/packages/app/e2e-tests/utils/sidebar.ts:53]: I'd suggest a less nebulous function name. Something like `assertConversationExists`, since that's what it seems to be doing

Comment (jrichter1) [reply workspaces/lightspeed/packages/app/e2e-tests/utils/sidebar.ts]: either works for me

**Review by karthikjeeyar — APPROVED** [review]
